src/cppgc.rs

```diff
@@ -0,0 +1,294 @@
+// Copyright 2019-2021 the Deno authors. All rights reserved. MIT license
+
+use crate::platform::Platform;
+use crate::support::int;
+use crate::support::Opaque;
+use crate::support::SharedRef;
+use crate::support::UniqueRef;
+
+extern "C" {
+  fn cppgc__initialize_process(platform: *mut Platform);
+  fn cppgc__shutdown_process();
+
+  fn cppgc__heap__create(
+    platform: *mut Platform,
+    wrappable_type_index: int,
+    wrappable_instance_index: int,
+    embedder_id_for_garbage_collected: u16,
+  ) -> *mut Heap;
+  fn cppgc__heap__DELETE(heap: *mut Heap);
+  fn cppgc__make_garbage_collectable(
+    heap: *mut Heap,
+    obj: *mut (),
+    trace: TraceFn,
+    destroy: DestroyFn,
+  ) -> *mut ();
+
+  fn cppgc__heap__enable_detached_garbage_collections_for_testing(
+    heap: *mut Heap,
+  );
+  fn cppgc__heap__collect_garbage_for_testing(
+    heap: *mut Heap,
+    stack_state: EmbedderStackState,
+  );
+
+  fn cppgc__visitor__trace(visitor: *const Visitor, member: *const ());
+}
+
+/// Process-global initialization of the garbage collector. Must be called before
+/// creating a Heap.
+///
+/// Can be called multiple times when paired with `ShutdownProcess()`.
+pub fn initalize_process(platform: SharedRef<Platform>) {
+  unsafe {
+    cppgc__initialize_process(&*platform as *const Platform as *mut _);
```

Comment:
**nitpick:** `&mut *platform as *mut Platform` perhaps? Or does `SharedRef` possibly have an `as_ptr` or `as_mut_ptr` API?

Reply:
SharedRef does not deref as mutable. This is a common pattern for passing pointers where Rust owernship does not matter:
https://github.com/denoland/rusty_v8/blob/12dca0cf0379f69a52bd710a4a64bf5efe9bcdc0/src/V8.rs#L188

Reply:
Unrelated to this PR, but we should probably have `as_ptr` and `as_mut_ptr` convenience methods on SharedRef to avoid the need to deref/ref and cast to pointer.